src/crush/CrushWrapper.cc

```diff
@@ -491,6 +491,69 @@ bool CrushWrapper::_bucket_is_in_use(int item)
   return false;
 }
 
+int CrushWrapper::_get_rules_with_class(list<pair<int, list<string>>> *rules)
```

Comment:
Uh oh... we have bigger problems.  The bucket id of the shadow trees cannot change or else data gets reshuffled.  Looking at rebuild_roots_with_classes(), that's exactly what it does (potentially renumber all shadow buckets). So... that's what needs to be fixed.  (And assuming we fix that, this problem of fixing the rules should go away.)

Let me take a closer look at this this afternoon.  :(